### ISSUE TITLE
convert.js does not add node-addon-api to the dependencies in binding.gyp

### ISSUE THREAD

**Issue by mcollina** [nodejs/node-addon-api#98]
As titled, after launching `covert.js`  the updated `binding.gyp` is not trying to build this specific module, resulting in missing symbols at runtime (in Node 6.11).

**Comment by mcollina** [comment]
I'm trying to see how do add this. Is it fine if I parse `binding.gyp`? I also need to add this module within the dependencies in `NAPI_CPP_EXCEPTIONS`.

**Comment by sampsongao** [comment]
Can you post an example of the output binding.gyp?

**Comment by mcollina** [comment]
Here it is:

```
{
  "targets": [
    {
      "target_name": "native-hdr-histogram",
      "cflags!": [ "-fno-exceptions" ],
      "cflags_cc!": [ "-fno-exceptions" ],
      "sources": [
        "src/hdr_encoding.h",
        "src/hdr_encoding.c",
        "src/hdr_histogram.h",
        "src/hdr_histogram.c",
        "src/hdr_histogram_log.h",
        "src/hdr_histogram_log.c",
        "hdr_histogram_wrap.cc",
        "histogram.cc"
      ],
      "dependencies": [
        "<(module_root_dir)/zlib/zlib.gyp:zlib"
      ],
      "include_dirs": [
        "<!@(node -p \"require('node-addon-api').include\")",
        "src/"
      ]
    }
  ]
}
```

What it's missing is the line in `dependencies`, like so:

```
{
  "targets": [
    {
      "target_name": "native-hdr-histogram",
      "cflags!": [ "-fno-exceptions" ],
      "cflags_cc!": [ "-fno-exceptions" ],
      "sources": [
        "src/hdr_encoding.h",
        "src/hdr_encoding.c",
        "src/hdr_histogram.h",
        "src/hdr_histogram.c",
        "src/hdr_histogram_log.h",
        "src/hdr_histogram_log.c",
        "hdr_histogram_wrap.cc",
        "histogram.cc"
      ],
      "dependencies": [
        "<(module_root_dir)/zlib/zlib.gyp:zlib",
        "<!(node -p \"require('node-addon-api').gyp\")"
      ],
      "include_dirs": [
        "<!@(node -p \"require('node-addon-api').include\")",
        "src/"
      ]
    }
  ]
}
```

**Comment by mhdawson** [comment]
@sampsongao have you had a chance to look at this ?

**Comment by sampsongao** [comment]
So currently the script find a <!(node -p "require('nan')") and replace it with node -p "require('node-addon-api').include"). Maybe we can do find the "dependencies" key and insert the include for node-addon-api and remove nan's if found.

**Comment by sampsongao** [comment]
@mcollina  can you try https://github.com/nodejs/node-addon-api/pull/158 to see if it work for your case?

**Comment by mhdawson** [comment]
ping @mcollina any chance you can try this out ?

**Comment by digitalinfinity** [comment]
@mcollina we believe this issue should have been resolved with #158- please reopen or file another issue if you still have problems here
